lib/NetPacksLib.cpp

```diff
@@ -921,7 +921,10 @@ DLL_LINKAGE void InsertNewStack::applyGs(CGameState *gs)
 	if(obj)
 		obj->putStack(slot, s);
 	else
+	{
 		logNetwork->error("[CRITICAL] InsertNewStack: invalid army object %d, possible game state corruption.", army.getNum());
+		delete s;

```

Comment:
For sure, we need to make some refactoring in future to use std::move for methods such as putStack instead of transferring memory ownership

Reply:
Can't we first check obj and if it is not null
auto s = new CStackInstance(type, count);

Reply:
So I wait for changes here.

Reply:
fixed it